fms/utils/serialization.py

```diff
@@ -288,29 +300,30 @@ def load_state_dict_into_model(
 
     # 3. Iterate over the weights and load them into the model
     used_keys = set()
-    sd_keys = list(state_dict.keys())
+    sd_keys = set(state_dict.keys())
+
     with torch.no_grad():
         for key in sd_keys:
             if key in used_keys:
                 continue
             used_keys.add(key)
-            try:
-                partial_sd = {key: state_dict[key]}
-                if partial_sd[key].device != initial_device:
-                    partial_sd[key] = partial_sd[key].to(device=initial_device)
-                fms_partial_sd = adapter(partial_sd)
-            except FusableWeightsMissingError as e:
-                for weight in e.missing_weights:
-                    used_keys.add(weight)
-                    partial_sd[weight] = state_dict[weight]
-                    if partial_sd[weight].device != initial_device:
-                        partial_sd[weight] = partial_sd[weight].to(
-                            device=initial_device
-                        )
-                fms_partial_sd = adapter(partial_sd)
+
+            partial_sd = {key: state_dict[key]}
+            # Find neighbors to the key. If the adapter requires a neighbor and
+            # this function doesn't find it, it will crash.
```

Comment:
might want to explain the purpose of "neighbor" in the comment ("e.g. if an adapter merges key/query/value it may need access to all elements of the state_dict at the same time")